main.go

```diff
@@ -351,3 +307,88 @@ func uploadFile(uploadURL string, archiveFilePath string) error {
 
 	return nil
 }
+
+func processStepResult(step *toolresults.Step) string {
+	outcome := step.Outcome.Summary
+
+	switch outcome {
+	case "success":
+		outcome = colorstring.Green(outcome)
+	case "failure":
+		if step.Outcome.FailureDetail != nil {
+			if step.Outcome.FailureDetail.Crashed {
+				outcome += "(Crashed)"
+			}
+			if step.Outcome.FailureDetail.NotInstalled {
+				outcome += "(NotInstalled)"
+			}
+			if step.Outcome.FailureDetail.OtherNativeCrash {
+				outcome += "(OtherNativeCrash)"
+			}
+			if step.Outcome.FailureDetail.TimedOut {
+				outcome += "(TimedOut)"
+			}
+			if step.Outcome.FailureDetail.UnableToCrawl {
+				outcome += "(UnableToCrawl)"
+			}
+		}
+		outcome = colorstring.Red(outcome)
+	case "inconclusive":
+		if step.Outcome.InconclusiveDetail != nil {
+			if step.Outcome.InconclusiveDetail.AbortedByUser {
+				outcome += "(AbortedByUser)"
+			}
+			if step.Outcome.InconclusiveDetail.InfrastructureFailure {
+				outcome += "(InfrastructureFailure)"
+			}
+		}
+		outcome = colorstring.Yellow(outcome)
+	case "skipped":
+		if step.Outcome.SkippedDetail != nil {
+			if step.Outcome.SkippedDetail.IncompatibleAppVersion {
+				outcome += "(IncompatibleAppVersion)"
+			}
+			if step.Outcome.SkippedDetail.IncompatibleArchitecture {
+				outcome += "(IncompatibleArchitecture)"
+			}
+			if step.Outcome.SkippedDetail.IncompatibleDevice {
+				outcome += "(IncompatibleDevice)"
+			}
+		}
+		outcome = colorstring.Blue(outcome)
+	}
+	return outcome
+}
+
+func GetSuccessOfExecution(steps []*toolresults.Step) (bool, error) {
+	outcomeByDimension, err := getOutcomeByDimension(steps)
+	if err != nil {
+		return false, err
+	}
+
+	for _, outcome := range outcomeByDimension {
+		if outcome.Summary != "success" {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func getOutcomeByDimension(steps []*toolresults.Step) (map[string]*toolresults.Outcome, error) {
+	groupedByDimension := make(map[string]*toolresults.Outcome)
+	for _, step := range steps {
+		key, err := json.Marshal(step.DimensionValue)
+		if err != nil {
+			return nil, err
+		}
+		if key != nil {
+			dimensionStr := string(key)
+			if groupedByDimension[dimensionStr] == nil || groupedByDimension[dimensionStr].Summary != "success" {
+				groupedByDimension[dimensionStr] = step.Outcome
```

Comment:
The previous iteration of this PR was being extra careful by only considering the last executed Step for a dimension. Turns out the API doesn't actually include any info about when the step was run in the payload! So instead, this code is simply accepting that if any of the Steps for a dimension were successful, we can count the dimension as successful. This is reasonable to do because it is nearly certain that a successful result is the final attempt--under what scenario would there be a retry if that isn't the case?

Reply:
This makes sense if the assumption is true. I am not fully aware what you can do on Android but because Apple and Google copy ideas of each other I want to highlight that Xcode has a test running mode called `Up until maximum repetitions`. In this mode it will retry the test as many times as the user requested and even the successful ones. In this mode the result of the last execution counts. 

Can you double check that this is not possible here?